### PR TITLE
In pending-approval-by:me box, offer gate chips instead of the approval stamp icon

### PR DESCRIPTION
**Pull request by jrobbins**
This is progress on issue #2333.

Currently we have an approval stamp icon that will open the approvals dialog showing all approvals.  This PR replaces the stamp icon with a set of approval chips that surface the state of each individual approval.   This will be helpful when we have more kinds of approvals, but for now we have at most one kind of approval per stage, so it is just a matter of users clicking a chip instead of an icon.  The chip currently opens the same approvals dialog, but that will change in the future.

In this PR:
* Change from loading ApprovalConfig objects to loading Gate objects.  The main difference is that Gate's also have a state that summarizes the outcome of Votes on the server, instead of the old way of dealing with the (non-summarized) Approval objects directly.
* Rip out a bunch of textual "highlights" info that could be surfaces in each feature row.  A lot of this was never fully implemented, and the parts that were turned out not to be useful to users.
* Remove the approval stamp icon from these feature rows.
* Display all the gate chips for any state that has an active gate.
* Instead of the approval stamp icon click creating an event that then opens the approval dialog, the gate chips create that same event and it has the same effect.
* Add code to cs-client and main.py to allow calling the existing API for Gates.

### CODE REVIEW

File: client-src/elements/chromedash-feature-row.js [hunk 1]
```diff
@@ -184,76 +142,53 @@ class ChromedashFeatureRow extends LitElement {
...
+    const activeGates = featureGates.filter(g => this.isActiveGate(g));
+    const activeStageIds = new Set(activeGates.map(g => g.stage_id));
+    const activeStagesAndTheirGates = [];
+    for (const activeStageId of activeStageIds) {
+      activeStagesAndTheirGates.push({
+        stageId: activeStageId,
+        gates: featureGates.filter(g => g.stage_id == activeStageId),
+      });
+    }
+    console.log(activeStagesAndTheirGates);
```
Comment (KyleJu) [inline client-src/elements/chromedash-feature-row.js:160]: Nit: more descriptive logging, e.g. console.log(activeStagesAndTheirGates is ...)

File: client-src/elements/chromedash-feature-row.js [hunk 2]
```diff
@@ -184,76 +142,53 @@ class ChromedashFeatureRow extends LitElement {
...
-      }
-    }
-    return result;
+  renderActiveStageAndGates(stageAndGates) {
+    const stageName = ''; // TODO(jrobbins) get this data.
+    return html`
+      <div>
+        ${stageName}
+        ${stageAndGates.gates.map(gate => html`
+        <chromedash-gate-chip
```
Comment (KyleJu) [inline client-src/elements/chromedash-feature-row.js:170]: Is this the correct indentation? e.g.
```suggestion
          <chromedash-gate-chip
```

File: client-src/elements/chromedash-feature-row.js [hunk 3]
```diff
@@ -184,76 +142,53 @@ class ChromedashFeatureRow extends LitElement {
...
-              Owners: ${owners.join(', ')}
-            </div>
-            ` : nothing}
-          ${activeApprovals && activeApprovals.length > 0 ? html`
-            <div>
-              ${this.renderApprovalsSoFar(activeApprovals)}
-            </div>
-            ` : nothing}
+          ${activeStages.length > 0 ? html`
+          <div>
```
Comment (KyleJu) [inline client-src/elements/chromedash-feature-row.js:186]: Correct HTML indentation here?
```suggestion
            <div>
```

File: client-src/elements/chromedash-gate-chip.js [hunk 4]
```diff
@@ -93,6 +99,23 @@ class ChromedashGateChip extends LitElement {
...
+    const event = new CustomEvent(eventName, {
+      bubbles: true,
+      composed: true,
+      detail,
+    });
+    this.dispatchEvent(event);
+  }
+
+  openApprovalsDialog() {
+    // handled in chromedash-myfeatures-page.js
```
Comment (KyleJu) [inline client-src/elements/chromedash-gate-chip.js:112]:
```suggestion
    // Handled in chromedash-myfeatures-page.js
```

File: client-src/elements/chromedash-gate-chip.js [hunk 5]
```diff
@@ -93,6 +99,23 @@ class ChromedashGateChip extends LitElement {
     `];
   }
 
+  _fireEvent(eventName, detail) {
```
Comment (KyleJu) [inline client-src/elements/chromedash-gate-chip.js:102]: nit: Could move this to a utility class in a follow-up. This is defined in chromedash-feature-row.js as well

File: client-src/js-src/cs-client.js [hunk 6]
```diff
@@ -198,6 +198,11 @@ class ChromeStatusClient {
         });
   }
 
+  getGates(featureId) {
+    return this.doGet(`/features/${featureId}/gates`);
+  }
+
+
```
Comment (KyleJu) [inline client-src/js-src/cs-client.js:205]:
```suggestion
```

**Review by KyleJu — APPROVED** [review]
LGMT with some nits

File: client-src/elements/chromedash-feature-row.js (hunk 1 above)
Comment (jrobbins) [reply client-src/elements/chromedash-feature-row.js:160]: Removed.  I had left that in by mistake.

File: client-src/elements/chromedash-feature-row.js (hunk 2 above)
Comment (jrobbins) [reply client-src/elements/chromedash-feature-row.js:170]: Fixed.

File: client-src/elements/chromedash-feature-row.js (hunk 3 above)
Comment (jrobbins) [reply client-src/elements/chromedash-feature-row.js:186]: Fixed.

File: client-src/elements/chromedash-gate-chip.js (hunk 4 above)
Comment (jrobbins) [reply client-src/elements/chromedash-gate-chip.js:112]: Fixed.

File: client-src/elements/chromedash-gate-chip.js (hunk 5 above)
Comment (jrobbins) [reply client-src/elements/chromedash-gate-chip.js:102]: Will do.

File: client-src/elements/chromedash-gate-chip.js [hunk 7]
```diff
@@ -93,6 +99,23 @@ class ChromedashGateChip extends LitElement {
...
+  }
+
+  openApprovalsDialog() {
+    // handled in chromedash-myfeatures-page.js
+    this._fireEvent('open-approvals-event', {
+      feature: this.feature,
+    });
+  }
+
+
```
[the remark this answers is not in the document]
Comment (jrobbins) [reply client-src/elements/chromedash-gate-chip.js:118]: Done.

File: client-src/js-src/cs-client.js (hunk 6 above)
Comment (jrobbins) [reply client-src/js-src/cs-client.js:205]: Done

**Pull request merged**